scmrepo/git/objects.py

```diff
@@ -136,13 +136,21 @@ def walk(self, top: tuple, topdown: Optional[bool] = True):
             yield top, dirs, nondirs
 
     def info(self, key: tuple) -> dict:
+        from scmrepo.utils import LazyDict
+
         obj = self.trie[key]
-        return {
-            "size": obj.size,
-            "type": "directory" if stat.S_ISDIR(obj.mode) else "file",
-            "sha": obj.sha,
-            "mode": obj.mode,
-        }
+
+        def size():
+            return obj.size
+
+        return LazyDict(
```

Comment:
Obviously, we could create a flag for `info` and then override `fs.walk`(and some others) to use it, but I wanted to try out this lazy approach, to see if it will even work in this setup. Looks like it does, but I might go back to a more explicit approach anyways (will get back to this in the morning).

Reply:
Using a flag is somewhat too explicit and requires overriding a lot of methods. Let's try the `LazyDict` for now and see how that goes.